src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -649,11 +648,16 @@ private void ValidateNonOriginHostHeader(string hostText)
             if (hostText != _absoluteRequestTarget!.Authority)
             {
                 if (!_absoluteRequestTarget.IsDefaultPort
-                    || hostText != _absoluteRequestTarget.Authority + ":" + _absoluteRequestTarget.Port.ToString(CultureInfo.InvariantCulture))
+                    || hostText != $"{_absoluteRequestTarget.Authority}:{_absoluteRequestTarget.Port}")
```

Comment:
Unrelated to the bug, but this temporary string allocation seems like it could be optimized away.

Reply:
https://github.com/dotnet/runtime/issues/110427

Reply:
We could do it ourselves. e.g. check that content before `:` matches `_absoluteRequestTarget.Authority` and content after `:` parses to an int and matches `_absoluteRequestTarget.Port`.

But if this path isn't that hot then probably not worth the complexity.

Reply:
Does it run per request ?

Reply:
If you're using this option and the host doesn't match the authority.